apps/api/app/agents/middleware/factory.py

```
@@ -0,0 +1,261 @@
+    subagent_tools: list[BaseTool] | None = None,
+    subagent_registry: Mapping[str, BaseTool] | None = None,
+    subagent_excluded_tools: set[str] | None = None,
+    subagent_tool_space: str = "general",
+    subagent_tool_runtime_config: ToolRuntimeConfig | None = None,
+    summarization_trigger: tuple = ("fraction", SUMMARIZATION_TRIGGER_FRACTION),
+    summarization_keep: tuple = ("tokens", SUMMARIZATION_KEEP_TOKENS),
+    compaction_threshold: float = COMPACTION_THRESHOLD,
+    max_output_chars: int = MAX_OUTPUT_CHARS,
+    vfs_enabled: bool = True,
+    compaction_excluded_tools: set[str] | None = None,
+    summarization_excluded_tools: set[str] | None = None,
```

Ruff (0.15.2)

[warning] Unused function argument: `summarization_excluded_tools`

(ARG001)